mobile/android/android-application-manager.ts

```diff
@@ -44,10 +44,35 @@ export class AndroidApplicationManager extends ApplicationManagerBase {
 	}
 
 	public async startApplication(appIdentifier: string): Promise<void> {
-		await this.adb.executeShellCommand(["monkey",
-			"-p", appIdentifier,
-			"-c", "android.intent.category.LAUNCHER",
-			"1"]);
+
+		/*
+		Example "pm dump <app_identifier> | grep -A 1 MAIN" output"
+			android.intent.action.MAIN:
+			3b2df03 org.nativescript.cliapp/com.tns.NativeScriptActivity filter 50dd82e
+			Action: "android.intent.action.MAIN"
+			Category: "android.intent.category.LAUNCHER"
+			--
+			intent={act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=org.nativescript.cliapp/com.tns.NativeScriptActivity}
+			realActivity=org.nativescript.cliapp/com.tns.NativeScriptActivity
+			--
+			Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=org.nativescript.cliapp/com.tns.NativeScriptActivity }
+			frontOfTask=true task=TaskRecord{fe592ac #449 A=org.nativescript.cliapp U=0 StackId=1 sz=1}
+		*/
+		const pmDumpOutput = await this.adb.executeShellCommand(["pm", "dump", appIdentifier, "|", "grep", "-A", "1", "MAIN"]);
```

Comment:
[Minor] Do we really need this `grep`? First of all, is `grep -A` available on all Android devices? Also we are getting part of the output of `pm dump` here and later we are searching for something via regular expression. I'm wondering, is it possible to make the regular expression smart enough to parse the full result.